tap_shiphero/discover.py

```diff
@@ -0,0 +1,79 @@
+import os
+import json
+from copy import deepcopy
+
+from singer.catalog import Catalog, CatalogEntry, Schema
+
+SCHEMAS = None
+FIELD_METADATA = None
+
+PKS = {
+    'orders': ['id'],
+    'products': ['id'],
+    'vendors': ['vendor_id'],
+    'shipments': ['shipment_id']
+}
+
+def get_abs_path(path):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
+
+def get_schemas():
+    global SCHEMAS, FIELD_METADATA
+
+    if SCHEMAS:
+        return SCHEMAS, FIELD_METADATA
+
+    SCHEMAS = {}
+    FIELD_METADATA = {}
+
+    schemas_path = get_abs_path('schemas')
+
+    file_names = [f for f in os.listdir(schemas_path)
+                  if os.path.isfile(os.path.join(schemas_path, f))]
+
+    file_names = sorted(file_names)
+
+    for file_name in file_names:
+        if file_name.endswith('.json'):
+            stream_name = file_name[:-5]
+            with open(os.path.join(schemas_path, file_name)) as data_file:
+                schema = json.load(data_file)
+
+            SCHEMAS[stream_name] = schema
+            pk = PKS[stream_name]
+            metadata = []
+
+            for prop, json_schema in schema['properties'].items():
+                if prop in pk:
+                    inclusion = 'automatic'
+                else:
+                    inclusion = 'available'
+                metadata.append({
```

Comment:
I'd still like to see this using the singer metadata code to keep this consistently correct.